pandas/util/testing.py

```diff
@@ -1185,8 +1201,13 @@ def assert_series_equal(
                 right._internal_get_values(),
                 check_dtype=check_dtype,
             )
-    elif is_interval_dtype(left) or is_interval_dtype(right):
-        assert_interval_array_equal(left.array, right.array)
+    elif is_interval_dtype(left) or is_interval_dtype(left):
+        # must cast to interval dtype to keep mypy happy
+        assert is_interval_dtype(right)
+        assert is_interval_dtype(left)
+        left_array = IntervalArray(left.array)
```

Comment:
yeah we don't want to do this (cast is ok), but don't actually coerce with a constructor.

Reply:
Ha got it, wasn't aware of typing.cast. Have updated the code.